app/models/internal_note.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class InternalNote < ApplicationRecord
+  belongs_to :document
+  belongs_to :user, optional: true
+  belongs_to :timeline_entry, class_name: "TimelineEntry", foreign_key: :timeline_entries_id, inverse_of: :internal_note
```

Comment:
I think the class_name and foreign_key options are likely superfluous here.

Reply:
Without these rails doesn't recognise the foreign key for timeline_entry when creating a InternalNote as i believe rails is expecting timeline_entry_id instead of timeline_entries_id

Reply:
D'oh that's a pain - makes sense to have them then